app/controllers/application_controller.rb

```diff
@@ -2618,4 +2618,12 @@ def restful?
     false
   end
   public :restful?
+
+  def determine_record_id_for_presenter
+    if @in_a_form
+      @edit && @edit[:rec_id]
+    else
+      @record.try(:id)
```

Comment:
Could you change this to `try!`: [docs](http://api.rubyonrails.org/v4.2.5/classes/Object.html#method-i-try-21)

```ruby
# @record.try(:id)
if @record.nil?
  nil
elsif !@record.respond_to?(:id)
  nil
else
  @record.id
end

# @record.try!(:id)
if @record.nil?
  nil
else
  @record.id
end
```

so `try!` is more similar to ruby's up and coming `@record.?id`